sycl/include/sycl/vector_preview.hpp

```diff
@@ -1032,41 +1025,87 @@ template <typename Type, int NumElements> class vec {
     return Lhs;                                                                \
   }                                                                            \
-  template <int Num = NumElements>                                             \
-  friend typename std::enable_if_t<Num != 1, vec &> operator OPASSIGN(         \
-      vec & Lhs, const DataT & Rhs) {                                          \
+  template <int Num = NumElements, typename T = DataT>                         \
+  friend typename std::enable_if_t<(Num != 1) && (COND), vec &>                \
+  operator OPASSIGN(vec & Lhs, const DataT & Rhs) {                            \
     Lhs = Lhs BINOP vec(Rhs);                                                  \
     return Lhs;                                                                \
   }
 
-#endif // __SYCL_USE_EXT_VECTOR_TYPE__
-
-  __SYCL_BINOP(+, +=, true)
-  __SYCL_BINOP(-, -=, true)
-  __SYCL_BINOP(*, *=, false)
-  __SYCL_BINOP(/, /=, false)
-
-  // TODO: The following OPs are available only when: DataT != cl_float &&
-  // DataT != cl_double && DataT != cl_half
-  __SYCL_BINOP(%, %=, false)
-  __SYCL_BINOP(|, |=, false)
-  __SYCL_BINOP(&, &=, false)
-  __SYCL_BINOP(^, ^=, false)
-  __SYCL_BINOP(>>, >>=, false)
-  __SYCL_BINOP(<<, <<=, true)
+  // std::byte is not an arithmetic type.
+  __SYCL_BINOP(+, +=, true, (!detail::is_byte_v<T>))
+  __SYCL_BINOP(-, -=, true, (!detail::is_byte_v<T>))
+  __SYCL_BINOP(*, *=, false, (!detail::is_byte_v<T>))
+  __SYCL_BINOP(/, /=, false, (!detail::is_byte_v<T>))
+
+  // The following OPs are available only when: DataT != cl_float &&
+  // DataT != cl_double && DataT != cl_half && DataT != BF16.
+  __SYCL_BINOP(%, %=, false,
+               (!detail::is_vgenfloat_v<T> && (!detail::is_byte_v<T>)))
+  // Bitwise operations are allowed for std::byte.
+  __SYCL_BINOP(|, |=, false, (!detail::is_vgenfloat_v<T>))
+  __SYCL_BINOP(&, &=, false, (!detail::is_vgenfloat_v<T>))
+  __SYCL_BINOP(^, ^=, false, (!detail::is_vgenfloat_v<T>))
+  __SYCL_BINOP(>>, >>=, false,
+               (!detail::is_vgenfloat_v<T> && (!detail::is_byte_v<T>)))
+  __SYCL_BINOP(<<, <<=, true,
+               (!detail::is_vgenfloat_v<T> && (!detail::is_byte_v<T>)))
+
+#undef BINOP_BASE
 #undef __SYCL_BINOP
-#undef __SYCL_BINOP_HELP
+
+  // Special <<, >> operators for std::byte.
+  // std::byte is not an arithmetic type and it only supports the following
+  // overloads of >> and << operators.
+  //
+  // 1 template <class IntegerType>
+  //   constexpr std::byte operator<<( std::byte b, IntegerType shift )
+  //   noexcept;
+  // 2 template <class IntegerType>
+  //   constexpr std::byte operator>>( std::byte b, IntegerType shift )
+  //   noexcept;
+#define __SYCL_SHIFT_BYTE(OP, OPASSIGN)                                        \
+  template <typename T = DataT>                                                \
+  friend typename std::enable_if_t<(detail::is_byte_v<T>), vec> operator OP(   \
```

Comment:
`friend typename std::enable_if_t<...`

I think `typename` is unnecessary here in other similar places.